### PR TITLE
8334733: Remove obsolete @enablePreview from tests after JDK-8334714

### PR DESCRIPTION
**Pull request by liach**
Remove the redundant `@enablePreview` and `--enable-preview` flags for enabling ClassFile API in the tests.  The remainder of these flags in all tests seem to serve preview APIs (such as ScopedValue) or language features (primitive pattern, module imports, etc.), or testing the enable preview flag itself.  Now there is fewer than 100 `@enablePreview` in the `test` directory.

To reviewers, there are some redundant changes and notes:

- There's one security test that used `ModuleInfoWriter` that depends on ClassFile API.
- Removed unnecessary exports of `jdk.internal.classfile.impl`. Remaining uses are:
  - `BoundAttribute::payloadLen` for javac attribute tests
  - Annotation reading/writing for javac annotation tests
- Line number changes to:
  - test/langtools/tools/javac/linenumbers/NestedLineNumberTest.java
  - test/langtools/tools/javac/linenumbers/NullCheckLineNumberTest.java
- Move from legacy jdk.internal.classfile to java.lang.classfile in:
  - test/langtools/tools/javac/NoStringToLower.java and 
  - test/langtools/tools/javac/T8003967/DetectMutableStaticFields.java
- Weird annotation processor behavior in test/langtools/tools/javac/annotations/parameter/ParameterAnnotations.java

  - Without preview and using explicit file name, the javac task fails; using the builder live AP object works both with and without preview.

Testing: tier 1-5.  Please inform me if any of these tests belong to higher tiers.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8334733](https://bugs.openjdk.org/browse/JDK-8334733): Remove obsolete @enablePreview from tests after JDK-8334714 (**Task** - P4)

### Reviewers
 * [Mandy Chung](https://openjdk.org/census#mchung) (@mlchung - **Reviewer**)
 * [Adam Sotona](https://openjdk.org/census#asotona) (@asotona - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22420/head:pull/22420` \
`$ git checkout pull/22420`

Update a local copy of the PR: \
`$ git checkout pull/22420` \
`$ git pull https://git.openjdk.org/jdk.git pull/22420/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22420`

View PR using the GUI difftool: \
`$ git pr show -t 22420`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22420.diff">https://git.openjdk.org/jdk/pull/22420.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22420#issuecomment-2504967528)
</details>

### CODE REVIEW

**Comment by asotona** [comment]
I would give a go to all trivial removals of `@enablePreview` after a brief check the tests pass.
However all the more complex changes potentially affecting various parts of the JDK infrastructure require deeper review and probably involve more reviewers.
My recommendation for easier review would be to split this 360-files patch into a patch with trivial removals of `@enablePreview` and one or more patches focused on particular area.

**Comment by liach** [comment]
I think the simple removal of `@enablePreview` already require multiple areas of engineers, so to involve a bit more changes to fix them throughoutly is fine; these are all test-only, so we can wait and this can go in after RDP1 began.

File: test/langtools/tools/javac/annotations/parameter/ParameterAnnotations.java [hunk 1]
```diff
@@ -641,8 +640,8 @@ private void doTest(Path base, String code, String binaryNameToCheck,
         }
 
         Task.Result result = new JavacTask(tb)
+                .processors(new TestAP())
```
Comment (mlchung) [inline test/langtools/tools/javac/annotations/parameter/ParameterAnnotations.java:643]: I assume this fix does not require this change, right?  If so, better to keep the original code and do this cleanup as a separate issue.

**Review by mlchung — COMMENTED** [review]
The change looks okay to me except `ParameterAnnotations.java`.   This task is part of JEP 484 and test-only.   It seems good to backport to 24.

File: test/langtools/tools/javac/annotations/parameter/ParameterAnnotations.java (hunk 1 above)
Comment (liach) [reply test/langtools/tools/javac/annotations/parameter/ParameterAnnotations.java:643]: It is required.  If we use the command line and FQN to specify a processor like in the current code, this test fails with some classpath error.

> Weird annotation processor behavior in test/langtools/tools/javac/annotations/parameter/ParameterAnnotations.java

>Without preview and using explicit file name, the javac task fails; using the builder live AP object works both with and without preview.

I should attach the exact error message:

```
test: testInnerClass
[DIRECT]:
- compiler.err.proc.processor.not.found: ParameterAnnotations$TestAP
- compiler.err.proc.no.explicit.annotation.processing.requested: T$I
2 errors
Exception running test testInnerClass: toolbox.Task$TaskError: Task javac failed: rc=1
toolbox.Task$TaskError: Task javac failed: rc=1
        at toolbox.AbstractTask.checkExit(AbstractTask.java:154)
        at toolbox.JavacTask.run(JavacTask.java:381)
        at toolbox.AbstractTask.run(AbstractTask.java:102)
        at toolbox.JavacTask.run(JavacTask.java:52)
        at toolbox.JavacTask.run(JavacTask.java:321)
        at ParameterAnnotations.doTest(ParameterAnnotations.java:650)
        at ParameterAnnotations.testInnerClass(ParameterAnnotations.java:151)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
        at java.base/java.lang.reflect.Method.invoke(Method.java:565)
        at toolbox.TestRunner.runTests(TestRunner.java:91)
        at ParameterAnnotations.runTests(ParameterAnnotations.java:82)
        at ParameterAnnotations.main(ParameterAnnotations.java:73)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
        at java.base/java.lang.reflect.Method.invoke(Method.java:565)
        at com.sun.javatest.regtest.agent.MainActionHelper$AgentVMRunnable.run(MainActionHelper.java:333)
        at java.base/java.lang.Thread.run(Thread.java:1447)
```

Same for all 5 tests in this file.

**Review by mlchung — APPROVED** [review]
Thanks for filing the issue to follow up.

**Review by asotona — APPROVED** [review]
Looks good to me, thanks.

**Comment by liach** [comment]
Thanks for the reviews!

/integrate

**Comment by liach** [comment]
/backport :jdk24

From internal discussion, this task is logically associated with the finalization of ClassFile API and is best done for the same release 24.

**Pull request closed without merging**